### DISCUSSION TITLE
iLogtail重命名为LoongCollector和目录布局变更

### DISCUSSION THREAD

**Discussion by yyuuttaaoo** [alibaba/ilogtail#1687] [Announcements]
[iLogtail 开源两周年：感恩遇见，畅想未来](https://mp.weixin.qq.com/s/RoFjoYlPLG1yOzDGc7vqIQ)
![LoongCollector](https://mmbiz.qpic.cn/mmbiz_png/yvBJb5IiafvlfRMKYLNRZdxyEred5wKAWVuZIpExTmnZdOAFh7ZX3hTNR9A1YE4EKTdGvbZgOgPb39RurH4M1BA/640?wx_fmt=png&from=appmsg&tp=webp&wxfrom=5&wx_lazy=1&wx_co=1)  

作为2024年RoadMap的一部分，iLogtail计划更名为LoongCollector，该计划将包含如下几个阶段：

- iLogtail二进制产物更名LoongCollector
- iLogtail目录和配置迁移到LoongCollector新布局
- iLogtail GitHub仓库迁移LoongCollector

### Linux

假设LoongCollector安装在`/opt/loongcollector`

实际的 LoongCollector 可执行文件`/opt/loongcollector/loongcollector`

库文件：

- `/opt/loongcollector/libPluginAdapter.so` 

- `/opt/loongcollector/libPluginBase.so`

自带证书：`/opt/loongcollector/ca-bundle.crt`

**配置文件目录：**`/opt/loongcollector/conf`

日志配置文件：

- `/opt/loongcollector/conf/apsara_log_conf.json` 

- `/opt/loongcollector/conf/plugin_logger.xml`

标识配置文件：

- `/opt/loongcollector/conf/user_defined_id`

采集配置文件：`/opt/loongcollector/conf/continuous_pipeline_config`

进程级文件：`/opt/loongcollector/conf/instance_config`

**数据目录：**`/opt/loongcollector/data`

检查点：

- `/opt/loongcollector/data/go_plugin_checkpoint` 

- `/opt/loongcollector/data/exactly_once_checkpoint` 

- `/opt/loongcollector/data/file_check_point`

容器路径映射：`/opt/loongcollector/data/docker_path_config.json`

未发送数据：`/opt/loongcollector/data/send_buffer_file_xxxxxxxxxxxx`

Crash临时文件：`/opt/loongcollector/data/backtrace.dat`

**日志目录：**`/opt/loongcollector/log`

主要日志：`/opt/loongcollector/log/loongcollector.log`

Go插件日志：`/opt/loongcollector/log/go_plugin.log`

日志库初始化日志：`/opt/loongcollector/log/logger_initialization.log`

Profile日志：`/opt/loongcollector/log/snapshot`

**run目录：**`/opt/loongcollector/run`

Pid文件：`/opt/loongcollector/run/loongcollector.pid`

inotify日志：`/opt/loongcollector/run/inotify_watcher_dirs`

进程信息日志：`/opt/loongcollector/run/app_info.json`

```plaintext
/
└── /opt/loongcollector/
                       ├── loongcollector
                       ├── libPluginAdapter.so
                       ├── libPluginBase.so
                       ├── ca-bundle.crt
                       ├── plugins
                       │      └── custom plugins
                       ├── dump (used exclusively by the service_http_server input plugin)
                       ├── thirdparty
                       │      ├── jvm
                       │      └── telegraf
                       ├── conf/
                       │      ├── scripts
                       │      ├── apsara_log_conf.json
                       │      ├── plugin_logger.xml
                       │      ├── user_defined_id
                       │      ├── authorization.json
                       │      ├── continuous_pipeline_config/
                       │      │                 ├── local/
                       │      │                 │         └── collect_stdout.json
                       │      │                 └── remote/
                       │      │                           └── collect_file.json
                       │      └── instance_config/
                       │                        ├── local/
                       │                        │         ├── ebpf.json
                       │                        │         └── loongcollector_config.json
                       │                        └── remote/
                       │                                  ├── region.json
                       │                                  └── resource.json
                       ├── data/
                       │       ├── file_check_point
                       │       ├── exactly_once_checkpoint/
                       │       ├── go_plugin_checkpoint/
                       │       ├── docker_path_config.json
                       │       ├── send_buffer_file_xxxxxxxxxxxx
                       │       └── backtrace.dat
                       ├── log/
                       │       ├── loongcollector.log
                       │       ├── loongcollector.log.1
                       │       ├── go_plugin.log
                       │       ├── go_plugin.log.1
                       │       ├── logger_initialization.log
                       │       └── snapshot/
                       └── run/
                               ├── loongcollector.pid
                               ├── inotify_watcher_dirs
                               └── app_info.json
```

### Windows

假设LoongCollector安装在`C:\loongcollector`

**二进制文件目录：**`C:\loongcollector\bin`

实际的 iLogtail 可执行文件`C:\loongcollector\bin\loongcollector.exe`

守护进程：`C:\loongcollector\bin\loongcollector_daemon.exe`

库文件：`C:\loongcollector\bin\libPluginAdapter.so` `C:\loongcollector\bin\libPluginBase.so`

主要配置文件：`C:\loongcollector\bin\loongcollector_config.json`

**配置文件目录：**`C:\loongcollector\conf`

日志配置文件：`C:\loongcollector\conf\apsara_log_conf.json` `C:\loongcollector\conf\plugin_logger.xml`

标识配置文件：`C:\loongcollector\conf\users` `C:\loongcollector\conf\user_defined_id`

采集配置文件：`C:\loongcollector\conf\config` `C:\loongcollector\conf\user_config.d` `C:\loongcollector\conf\user_yaml_config.d`

自带证书：`C:\loongcollector\conf\ca-bundle.crt`

**数据目录：**`C:\loongcollector\data`

检查点：`C:\loongcollector\data\checkpoint` `C:\loongcollector\data\checkpoint_v2` `C:\loongcollector\data\logtail_check_point`

容器路径映射：`C:\loongcollector\data\docker_path_config.json`

未发送数据：`C:\loongcollector\data\logtail_buffer_file_xxxxxxxxxxxx`

**日志目录：**`C:\loongcollector\logs`

主要日志：`C:\loongcollector\logs\loongcollector.LOG` `C:\loongcollector\logs\loongcollector.n.LOG`

Go插件日志：`C:\loongcollector\logs\go_plugin.LOG` `C:\loongcollector\logs\go_plugin.LOG.n`

日志库初始化日志：`C:\loongcollector\logs\logger_initialization.log`

Profile日志：`C:\loongcollector\logs\snapshot`

inotify日志：`C:\loongcollector\logs\inotify_watcher_dirs`

进程信息日志：`C:\loongcollector\logs\app_info.json`

### 升级问题

新的目录结构固然清晰，但带来的问题是与以前配置目录和数据目录不兼容了，升级需要迁移才能保证采集连续性。

配置和数据目录不同，数据目录必须保证连续性，因为随时在变，而配置目录可以认为大部分需求都是在配置不变的情况下要进行升级、回滚。

开源版需要升级脚本，将原ilogtail目录下的数据copy到新的目录中。

## iLogtail GitHub仓库迁移LoongCollector

1. 将iLogtail代码库rename为LoongCollector
2. 新建iLogtail代码库，指向LoongCollector

**Comment by quzard** [comment]
`loongcollector_config.json` 需要和 `loongcollector` 放在同个目录里，为了能够自定义其他目录
